Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3061,6 +3061,29 @@ extension BrowserViewController: PreferencesObserver {
             setupTabs()
             updateTabsBarVisibility()
             updateApplicationShortcuts()
+            
+            // Destroy all tab data
+            tabManager.allTabs.forEach({
+                tabManager.removeAllBrowsingDataForTab($0)
```

Comment:
We need not do this since cookie and cache clearable already does that for us

Reply:
https://github.com/brave/brave-ios/blob/cbe5efa082bb453ed7b389391f72ab5941a32cdd/Client/Frontend/Settings/Clearables.swift#L52

Reply:
Also let’s make a check to delete data if switching from normal to private, since we already have a logic to clear data when going to normal from private

Reply:
SettingsController doesn't have access to `tabManager` so it won't be able to remove all the tabs or reset configuration.

For the logic for going from `normal to private`, the code above does that because any time you toggle the switch, it will clear the data regardless of whether it is true or false.